### ISSUE TITLE
Extra datasets for scikit-learn

### ISSUE THREAD

**Issue by ghost** [scikit-learn-contrib/scikit-learn-contrib#9]
Request for project inclusion in scikit-learn-contrib
- Project name: extra-datasets
- Project description: Extra datasets for scikit-learn.
- Authors: https://github.com/3d20faaf7d
- Current repository: https://github.com/3d20faaf7d/extra-datasets
- Requirements:
  - [ ] scikit-learn compatible (check_estimator passed)
  - [x] Documentation (guide, API reference, example gallery)
  - [x] Unit tests (coverage: xx%)
  - [x] Python3 compatible
  - [x] PEP8 compliant
  - [x] Continuous integration

Notes:
- there was no need to use _check_estimator_ in my case
- there are 2-3 PEP8 warnings (e.g. E501 line too long)
- Continuous integration on Travis and Circle CI
- I have also added a tiny script _ci_scripts/local_check.py_ to check PEP8, PEP257, doctests and so on (locally).

I will add more dataset wrappers in the following period.

**Comment by amueller** [comment]
I feel like openml.org and the python api for them might be a better solution to the dataset problem. What do you think?

**Comment by mfeurer** [comment]
Which you can find [here](https://github.com/openml/openml-python). Dataset loading is already quite stable.

**Comment by amueller** [comment]
@mfeurer wow you're on top of things ;) <dadjoke> one might even say on feuer </dadjoke>

**Comment by fabianp** [comment]
+1 from me. The documentation is done with care and package works as advertised.

Only thing, I would love to see more datasets :-)

**Comment by mblondel** [comment]
What is the project vision and how different is it from https://github.com/jaberg/skdata?

With only one dataset supported, I feel that it might be a bit too soon to join contrib.

**Comment by 3d20faaf7d** [comment]
I didn't do my homework well. I really didn't know about the existence of **skdata**.

About **openml** - it would be indeed better to have all the datasets in the world just there, all accessible via a uniform API, then work around the specifics of each dataset in tiny extra processing steps.

What I understand now:

* The implementation (not necessarily the interface) of the library should be changed to rely on openml (thus get rid of Google Drive download tricks and so on);
* There is no point of adding the library to **contrib** at the moment. Too soon for that (only one dataset);
* There is the option to include the dataset wrapper (and eventually other ones) in skdata project -  which has though the tiny disadvantage of not being owned by an organization).

Thank you!

**Comment by mfeurer** [comment]
One more note from my side about datasets. For most use cases, you actually don't want to use a dataset, but a task. A task contains a dataset, an evaluation measure and a split into train, valid and test set. This concept is implemented in both OpenML and skdata. Maybe it is helpful to implement a concept of tasks here as well if the additional datasets should not only be used for unit testing implementations of ML algorithms.

**Comment by glemaitre** [comment]
I am closing this proposal since that the repository does not exist anymore.
